statemachine/src/main/java/com/amazonaws/stepfunctions/cloudformation/statemachine/UpdateHandler.java

```diff
@@ -33,6 +33,8 @@
         metricsRecorder.setMetricsFromResourceModel(model);
 
         try {
+            verifyStateMachineArnIsPresent(model.getArn());
```

Comment:
nit: I think update handler needs similar in-progress marker as deleteHandler. This might be out of scope for this PR but I would recommend applying similar logic (mark "in-progress" before marking "success") as update state machine usually take longer than deletion.

Reply:
To clarify, the reason we need to add the in-progress logic to the delete handler is because the operation is asynchronous. The DeleteStateMachine API returns success after the state machine has been marked for deletion, but the CFN contract expects the handler to only return success once the resource is actually deleted. This is why we first return in progress, then poll for the state machine to actually be deleted before returning success. 

This does not apply to the UpdateStateMachine API as the call returns once the update is successful.